hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterJoinManager.java

```diff
@@ -759,7 +809,9 @@ private void startJoin() {
 
             } finally {
                 reset();
-                partitionService.resumeMigration();
+                if (shouldTriggerRepartition) {
```

Comment:
So if one member is restarting with persistence and another restarting without, the master node will pause migrations to both of them forever right? Unless there is another node later joining without persistence.

Reply:
Migrations are resumed as soon as the joining member with persistence finishes loading from disk in https://github.com/hazelcast/hazelcast-enterprise/pull/4124/files#diff-354d08a895ea0f0c909aac33f39bbd686ce203d17e9d394328c58d12a4454117R53-R57 .